src/test/java/networkbook/logic/commands/OpenEmailCommandTest.java

```diff
@@ -34,67 +34,74 @@ public class OpenEmailCommandTest {
-            throw new UnsupportedOperationException();
+            throw new AssertionError("This method should not be called.");
         }
 
         public void setGuiSettings(GuiSettings guiSettings) {
-            throw new UnsupportedOperationException();
+            throw new AssertionError("This method should not be called.");
         }
 
         public Path getNetworkBookFilePath() {
-            throw new UnsupportedOperationException();
+            throw new AssertionError("This method should not be called.");
         }
 
         public void setNetworkBookFilePath(Path networkBookFilePath) {
-            throw new UnsupportedOperationException();
+            throw new AssertionError("This method should not be called.");
         }
 
         public ReadOnlyNetworkBook getNetworkBook() {
             return TypicalPersons.getTypicalNetworkBook();
         }
 
         public void setNetworkBook(ReadOnlyNetworkBook networkBook) {
-            throw new UnsupportedOperationException();
+            throw new AssertionError("This method should not be called.");
         }
 
         public boolean hasPerson(Person person) {
-            throw new UnsupportedOperationException();
+            throw new AssertionError("This method should not be called.");
         }
 
         public void deletePerson(Person target) {
-            throw new UnsupportedOperationException();
+            throw new AssertionError("This method should not be called.");
         }
 
         public void addPerson(Person person) {
-            throw new UnsupportedOperationException();
+            throw new AssertionError("This method should not be called.");
         }
 
         public void setItem(Person target, Person edited) {
-            throw new UnsupportedOperationException();
+            throw new AssertionError("This method should not be called.");
         }
 
+        public boolean canUndoNetworkBook() {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        public boolean canRedoNetworkBook() {
+            throw new AssertionError("This method should not be called.");
+        }
         public void undoNetworkBook() {
-            throw new UnsupportedOperationException();
+            throw new AssertionError("This method should not be called.");
         }
 
         public void redoNetworkBook() {
-            throw new UnsupportedOperationException();
+            throw new AssertionError("This method should not be called.");
         }
 
         public boolean isValidLinkIndex(Index personIndex, Index linkIndex) {
-            throw new UnsupportedOperationException();
+            throw new AssertionError("This method should not be called.");
         }
 
         public Link openLink(Index personIndex, Index linkIndex) {
-            throw new UnsupportedOperationException();
+            throw new AssertionError("This method should not be called.");
```

Comment:
Please either:
* Keep the original implementation of using `UnsupportedOperationException`, or
* Attribute with `//@@author` on top of every method.